arduino-core/src/processing/app/Sketch.java

```diff
@@ -351,13 +351,22 @@ public void saveAs(File newFolder) throws IOException {
         file.saveAs(new File(newFolder, file.getFileName()));
     }
 
-    folder = newFolder;
 
     // Copy the data folder (this may take a while.. add progress bar?)
     if (getDataFolder().exists()) {
       File newDataFolder = new File(newFolder, "data");
+      // Create the data folder
+      if (!newDataFolder.mkdirs()) {
```

Comment:
We can check if the folder exists, but since we just created the parent folder at line 341, it is very unlikely the folder already exits.

Never the less it won't harm if we implement this extra check. So I will add the check and test if it still works.